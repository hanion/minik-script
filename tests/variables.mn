// declaration
a :: 3.14;
b := a * 2.7;
str := "mystring";

print(a); // 3.14
print(b); // 8.478
print(str + " test"); // mystring test
print(str); // mystring


// assignment
a = 1;
b = 2;
str = "newstr";
print(a); // 1
print(b); // 2
print(str); // newstr

// arithmetic
print(2 + 3 * 6);   // 20
print((2 + 3) * 6); // 30

print(6 / 3 * 2);   // 4
print(6 / (3 * 2)); // 1

{
	i := 0;
	++i;
	print(i == 1);   // true
	print(++i == 2); // true
	print(i == 2);   // true
}
{
	i := 0;
	--i;
	print(i == -1);   // true
	print(--i == -2); // true
	print(i == -2);   // true
}

// redefinition
print("should give redefinition error ->");
a := 2;
print(a);
