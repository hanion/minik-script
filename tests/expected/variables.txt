3.140000
8.478000
mystring test
mystring
1.000000
2.000000
newstr
20.000000
30.000000
4.000000
1.000000
true
true
true
true
true
true
should give redefinition error ->
[ERROR] [line 44], Redefinition of 'a'.
